3 lesson homework

Prokopyevt opened the pull request.

Commit:
Домашнее задание 3 урока.
 Basic Auth с Roles и Permissions с InMemory переделана на работу с таблицей пользователей в базе данных

Commit:
Домашнее задание 3 урока.
 Basic Auth с Roles и Permissions с InMemory переделана на работу с таблицей пользователей в базе данных

Korgutlova commented on src/main/java/com/simbirsoft/springcourse/config/SecurityConfig.java:
```diff
@@ -0,0 +1,44 @@
+            .httpBasic();
+    }
+
+
```
пустые строки убираем (форматирование)

Korgutlova commented on src/main/java/com/simbirsoft/springcourse/service/security/UserDetailsServiceImpl.java:
```diff
@@ -0,0 +1,27 @@
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+    @Service
```
форматирование слетело, отступ

Korgutlova commented on src/main/java/com/simbirsoft/springcourse/model/User.java:
```diff
@@ -0,0 +1,58 @@
+        this.authorities = authorities;
+    }
+
+    public Long getId() {
```
для удобства можно использовать lombok

ramilmmk replied:
> для удобства можно использовать lombok

об этом на следующем занятии)

Korgutlova reviewed (changes requested):
Так вроде требования с добавлением авторизации и ролей выполнены, однако в работе я не увидела permissions (разрешений)

ramilmmk commented on src/main/java/com/simbirsoft/springcourse/model/User.java:
```diff
@@ -0,0 +1,58 @@
+import java.util.Set;
+
+@Entity
+@Table(name = "userdata")
```
Может переименовать таблицу в "users"?

ramilmmk commented on src/main/java/com/simbirsoft/springcourse/repository/UserRepository.java:
```diff
@@ -0,0 +1,8 @@
+import com.simbirsoft.springcourse.model.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface UserRepository extends JpaRepository<User, Long> {
```
@Repository?

Commit:
Переправленное Домашнее задание 3 урока.
1.Переименована  таблица из "userdata" в "users"
2.Добавлен @repository в userRepository
3.Добавлены permissions

Commit:
Переправленное Домашнее задание 3 урока.
1.Переименована  таблица из "userdata" в "users"
2.Добавлен @repository в userRepository
3.Добавлены permissions
4. Удалены неиспользуемые импорты

Korgutlova commented on src/main/resources/db/migration/V2_0_0_1__userbase.sql:
```diff
@@ -16,14 +16,14 @@ CREATE TABLE IF NOT EXISTS user_authority
 
-INSERT INTO userdata (id, username, password) VALUES (1, 'Ivan', '$2y$12$GEiD1FfhRc22AqgF8kmJwe51nknKX4B9nTfIytCjJC4Ps9CViLGAy');
-INSERT INTO userdata (id, username, password) VALUES (2, 'Dima', '$2y$12$fsuEbjqH1mnRsIqJNajaMeW866kZh9a3ovNjvI9D8bm5.IeBoC.lq');
+INSERT INTO users (id, username, password) VALUES (1, 'Ivan', '$2y$12$GEiD1FfhRc22AqgF8kmJwe51nknKX4B9nTfIytCjJC4Ps9CViLGAy');
```
миграции не меняем, а добавляем новые  если что-то хотим поменять (даже если это название)  

на будущее надо запомнить

Korgutlova reviewed (approved):
Отлично, что добавил пермишены! Можешь мержить.
